frappe/public/js/frappe/desk.js

```diff
@@ -582,3 +584,139 @@ frappe.get_module = function (m, default_module) {
 
 	return module;
 };
+
+class TranslationsLoader {
+	constructor() {
+		// Build URL
+		const url = new URL(
+			"/api/method/frappe.translate.load_all_translations",
+			window.location.origin
+		);
+		url.searchParams.append("lang", frappe.boot.lang);
+		this.url = url;
+
+		// Check that translations need to be refreshed
+		const lastHash = localStorage.getItem("translations_hash");
+		const currHash = JSON.stringify(String(frappe.boot.translations_hash));
+		this.isStale = !lastHash || !currHash || lastHash?.length < 7 || lastHash !== currHash;
+		if (this.isStale && frappe.boot.developer_mode) {
+			console.info(`Stale translations: ${lastHash} -> ${currHash}`);
+		}
+
+		// https://developer.mozilla.org/en-US/docs/Web/API/CacheStorage
+		this.useLocalStorage = !window.caches?.open; // Use legacy if Caches API is not available
```

Comment:
Should we consider dropping this legacy support thing, I don't think it provides a lot of value.

Instead, on unsupported browsers (~4% [caniuse](https://caniuse.com/mdn-api_cache)), we could just fetch the values everytime.

Reply:
Wait, even better: just add the hash to the fetched URL, just like we do with assets... This feels so obvious now. Then add the `Cache-Control` header.